### ISSUE TITLE
Incorporating singularity build on local machine.

### ISSUE THREAD

**Issue by russelljjarvis** [russelljjarvis/neuronunit#13]
Hi @rgerkin and @sagpat117

Here is some documentation of singularity progress. 
# 1 build from source:
```
$ mkdir ~/git
$ cd ~/git
$ git clone https://github.com/singularityware/singularity.git
$ cd singularity
$ ./autogen.sh
```
Autogen did not work, even after supplying automake as described here, and using `sudo apt-get install automake`
https://geeksww.com/tutorials/libraries/m4/installation/installing_m4_macro_processor_ubuntu_linux.php
Installing the yum packages specified below
https://hpc.nih.gov/apps/singularity.html
in apt-get as opposed to yum would probably work, but no time. Skip to install approach # 2.
```
$ ./configure --prefix=/usr/local --sysconfdir=/etc
$ make
$ sudo make install
```
# 2 Workaround:
```
VERSION=2.2.1
wget https://github.com/singularityware/singularity/releases/download/$VERSION/singularity-$VERSION.tar.gz
tar xvf singularity-$VERSION.tar.gz
cd singularity-$VERSION
./configure --prefix=/usr/local
make
sudo make install
```
Alias to build a singularity image based on a local docker container:
```
alias d2s='sudo docker run -v /var/run/docker.sock:/var/run/docker.sock -v `pwd`/singularity\image:/output --privileged -t --rm singularityware/docker2singularity pnp:latest'


```

This results in a directory 
```
`pwd`/singularityimage/
```
containing the file:
```
pnp_latest-2017-04-10-e61582246138.img
```
# Enter the Sinularity image:
Then to enter the singularity vm (as opposed to docker container):
```
sudo singularity shell -w pnp_latest-2017-04-10-e61582246138.img 
Singularity: Invoking an interactive shell within container...

root@rjjarvis:/root# su jovyan
jovyan@rjjarvis:/root$ cd ~
jovyan@rjjarvis:~$ ls
```
Singularity image is build from local container, by entering a different special purpose singularity written docker container, whose only only function is to write a singularity image, based on the supplied docker container: 

Giving us an interactive session in our familiar docker like environment:
```
jovyan@rjjarvis:~$ ls
jLEMS     LEMS        NeuroML2    nrn-7.4        org.neuroml1.model  org.neuroml.import  org.neuroml.model.injectingplugin  sciunit  x86_64
jNeuroML  libNeuroML  neuronunit  openmpi-2.0.0  org.neuroml.export  org.neuroml.model   pylems                             work
```

**Comment by russelljjarvis** [comment]
Its not intended for solving the cluster problem its intended for local development. In my experience its easier to debug a program on a cluster if its almost the same on your local machine. I am trying to make my local machine match up as much of possible. The cluster program should only be different in scale.

It would be better to upload the docker image locally from my own machine, but I have not yet prepared or pushed an image to docker stacks that would be suitable for that yet. 

In summary its just a faster way to prototype a small scale version of what the cluster will eventually see.

**Comment by rgerkin** [comment]
Since we can use Zuse with Docker, there is no urgency in getting the Singularity build to work, so let's close this for now.
